DataLayer/Context/GameBoardDbContext.Model.cs

```diff
@@ -9,10 +11,47 @@ protected override void OnModelCreating(ModelBuilder builder)
         {
             base.OnModelCreating(builder);
 
-            builder.Entity<IdentityUser>(
-                entity => entity.ToTable("User"));
+            builder.Entity<ApplicationUser>(
+                entity =>
+                {
+                    entity.Property(e => e.UserName).HasMaxLength(16);
 
-            builder.Entity<IdentityRole>( 
+                    entity.ToTable("User");
+                });
+
+            builder.Entity<Friendship>(
+                entity =>
+                {
+                    entity.HasKey(e => e.Id);
+                    entity.Property(e => e.Id)
+                        .HasMaxLength(32)
+                        .IsRequired()
+                        .ValueGeneratedOnAdd();
+
+                    entity.Property(e => e.FriendshipStatus)
+                        .IsRequired();
+                    entity.Property(e => e.RequestedById)
+                        .IsRequired();
+                    entity.Property(e => e.RequestedToId)
+                        .IsRequired();
+
+                    entity.HasIndex(e => new { e.RequestedById, e.RequestedToId })
+                        .IsUnique();
+
+                    entity.HasOne(e => e.RequestedBy)
+                        .WithMany(u => u.SentRequests)
+                        .HasForeignKey(e => e.RequestedById)
+                        .OnDelete(DeleteBehavior.Restrict);
+                        //Can I set DeleteBehavior to NO ACTION?
```

Comment:
I'm pretty sure `DeleteBehavior.Restrict` causes `ON DELETE NO ACTION`. Remove the comment.

Reply:
Restrict does not let you perform an action that would result in the constraint violation. Thus, If I tried to delete my account (and I didn't have INSTEAD OF TRIGGER), I would receive an error (and my account wouldn't have been deleted). No action just ignores constraints. If I could set DeleteBehaviour to NO ACTION I could have AFTER DELETE TRIGGER instead of INSTEAD OF DELETE TRIGGER.

Reply:
But Restrict is the EF equivalent of `NO ACTION`. See [here](https://docs.microsoft.com/en-us/ef/core/saving/cascade-delete), 
> If you don't want EF Core to ever delete an entity automatically or null out the foreign key automatically, then use Restrict.
Also, you can take a look at the generated migration `.sql` script to confirm that the FKs have indeed `ON DELETE NO ACTION` constraints.